src/app/view/components/_common/iframe/service/dot-iframe-porlet-legacy-resolver.service.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { LoginService } from 'dotcms-js';
+import { LoginServiceMock, mockUser } from './../../../../../test/login-service.mock';
+import { RouterTestingModule } from '@angular/router/testing';
+import { async } from '@angular/core/testing';
+import { mockDotRenderedPage } from '../../../../../test/dot-page-render.mock';
+import { of } from 'rxjs';
+import { ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { DotPageStateService } from '@portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service';
+import { DOTTestBed } from '@tests/dot-test-bed';
+import { DotPageRenderService } from '@services/dot-page-render/dot-page-render.service';
+import { DotContentletLockerService } from '@services/dot-contentlet-locker/dot-contentlet-locker.service';
+import { DotPageRenderState, DotPageRender } from '@portlets/dot-edit-page/shared/models';
+import { DotIframePortletLegacyResolver } from './dot-iframe-porlet-legacy-resolver.service';
+import { DotLicenseService } from '@services/dot-license/dot-license.service';
+
+const route: any = jasmine.createSpyObj<ActivatedRouteSnapshot>('ActivatedRouteSnapshot', [
+    'toString'
+]);
+
+const state: any = jasmine.createSpyObj<RouterStateSnapshot>('RouterStateSnapshot', ['toString']);
+
+route.queryParams = {};
+
+describe('DotIframePorletLegacyResolver', () => {
+    let dotPageStateService: DotPageStateService;
+    let dotPageStateServiceRequestPageSpy: jasmine.Spy;
+    let resolver: DotIframePortletLegacyResolver;
+    let dotLicenseService: DotLicenseService;
+
+    beforeEach(async(() => {
+        const testbed = DOTTestBed.configureTestingModule({
+            providers: [
+                DotPageStateService,
+                DotIframePortletLegacyResolver,
+                DotPageRenderService,
+                DotContentletLockerService,
+                DotLicenseService,
+                {
+                    provide: ActivatedRouteSnapshot,
+                    useValue: route
+                },
+                {
+                    provide: RouterStateSnapshot,
+                    useValue: state
+                },
+                {
+                    provide: LoginService,
+                    useClass: LoginServiceMock
+                }
+            ],
+            imports: [RouterTestingModule]
+        });
+
+        dotPageStateService = testbed.get(DotPageStateService);
+        dotPageStateServiceRequestPageSpy = spyOn(dotPageStateService, 'requestPage');
+        resolver = testbed.get(DotIframePortletLegacyResolver);
+        dotLicenseService = testbed.get(DotLicenseService);
+        state.url = '/rules';
+    }));
+
+    it('should return a DotRenderedPageState', () => {
```

Comment:
This is not returning this.

This is why I mention that we should return the url for the iframe resolved from here because if not, this resolver turns to be just like a "canAccess" which is like a kinda guard, while the resolver resolve everything and just return the url to set in the iframe.

And with this approach you have to worried about logic in the component like the double flag we talked about.